### PR TITLE
docs: update span compression links

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR updates our span compression docs to match current agent support.

* Closes https://github.com/elastic/observability-docs/issues/2203.

### CODE REVIEW

File: docs/span-compression.asciidoc [hunk 1]
```diff
@@ -61,5 +61,28 @@ Spans with longer duration are not compressed. Please refer to the agent documen
...
+| {apm-java-ref-v}/config-huge-traces.html#config-span-compression-exact-match-max-duration[`span_compression_exact_match_max_duration`]
+// | **.NET agent**
+// | {apm-dotnet-ref-v}/config-span-compression-same-kind-max-duration.html[`SpanCompressionSameKindMaxDuration`]
+// | {apm-dotnet-ref-v}/config-span-compression-exact-match-max-duration.html[`SpanCompressionExactMatchMaxDuration`]
+| **Node.js agent**
+| {apm-node-ref-v}/configuration.html#span-compression-same-kind-max-duration[`spanCompressionSameKindMaxDuration`]
+| {apm-node-ref-v}/configuration.html#span-compression-exact-match-max-duration[`spanCompressionExactMatchMaxDuration`]
+// | **PHP agent**
+// | {apm-php-ref-v}[``]
+// | {apm-php-ref-v}[``]
```
Comment (bmorelli25) [inline docs/span-compression.asciidoc:81]: @SergeyKleyman are there docs for this feature? I wasn't able to find a PR that matches the issue (https://github.com/elastic/apm-agent-php/issues/425).

File: docs/span-compression.asciidoc [hunk 2]
```diff
@@ -61,5 +61,28 @@ Spans with longer duration are not compressed. Please refer to the agent documen
...
+| Agent | Same-kind config | Exact-match config
+| **Go agent**
+| {apm-go-ref-v}/configuration.html#config-span-compression-same-kind-duration[`ELASTIC_APM_SPAN_COMPRESSION_SAME_KIND_MAX_DURATION`]
+| {apm-go-ref-v}/configuration.html#config-span-compression-exact-match-duration[`ELASTIC_APM_SPAN_COMPRESSION_EXACT_MATCH_MAX_DURATION`]
+| **Java agent**
+| {apm-java-ref-v}/config-huge-traces.html#config-span-compression-same-kind-max-duration[`span_compression_same_kind_max_duration`]
+| {apm-java-ref-v}/config-huge-traces.html#config-span-compression-exact-match-max-duration[`span_compression_exact_match_max_duration`]
+// | **.NET agent**
+// | {apm-dotnet-ref-v}/config-span-compression-same-kind-max-duration.html[`SpanCompressionSameKindMaxDuration`]
+// | {apm-dotnet-ref-v}/config-span-compression-exact-match-max-duration.html[`SpanCompressionExactMatchMaxDuration`]
```
Comment (bmorelli25) [inline docs/span-compression.asciidoc:75]: @gregkalapos I found the docs for these, but they're not published yet. Would you like me to help with getting the newer versions of your docs published?

Comment (gregkalapos) [reply docs/span-compression.asciidoc:75]: @bmorelli25 oh, that'd be super nice! It's on my ToDo list, but if you just update it to the latest version `1.17.0`, it'd be awesome.

Comment (bmorelli25) [reply docs/span-compression.asciidoc:75]: Moving conversation to https://github.com/elastic/apm-agent-dotnet/issues/1842#issuecomment-1270661995.

Comment (bmorelli25) [inline docs/span-compression.asciidoc:75]: Uncomment out .NET links

```suggestion
| **.NET agent**
| {apm-dotnet-ref-v}/config-span-compression-same-kind-max-duration.html[`SpanCompressionSameKindMaxDuration`]
| {apm-dotnet-ref-v}/config-span-compression-exact-match-max-duration.html[`SpanCompressionExactMatchMaxDuration`]
```

Comment (bmorelli25) [reply docs/span-compression.asciidoc]: 1.17 docs are live

**Review by axw — APPROVED** [review]

**Pull request merged**